clang/include/clang/Basic/FixedPoint.h

```diff
@@ -165,7 +165,7 @@ class APFixedPoint {
   std::string toString() const {
     llvm::SmallString<40> S;
     toString(S);
-    return S.str();
+    return std::string(S.str());
```

Comment:
I added a std::string conversion operator for `SmallString` so you can do`std::string(S)` instead.

Reply:
Seems like there's quite a few of these. I think you should be able to find these cases as either `std::string\([^\)]*\.str\(\)\)` or `\.str\(\)\.str\(\)`.

Reply:
Is it ok if I address that in a follow up PR? Looks like those patches are present on `swift/master-next` but not on `swift/master`, so it would require cherry-picking more patches in this PR, further delaying the manual merge.

Reply:
Of course!